marketplaceItems/drawApp/draw_app.js

```diff
@@ -0,0 +1,504 @@
+//  Copyright 2018 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global  */
+
+(function() {
+    var DESKTOP_LINE_DISTANCE_FROM_AVATAR_M = 1;
+    
+    
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
+    var button = tablet.addButton({
+        text: 'DRAW',
+        icon: Script.resolvePath('Assets/Icons/draw-i.svg'),
+        activeIcon: Script.resolvePath('Assets/Icons/draw-a.svg')
+    });
+  
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+
+    /* PLAY A SOUND: Plays the specified sound at the position of the user's Avatar using the volume and playback 
+    mode requested. */
+    var injector;
+    function playSound(sound, volume, position, localOnly){
+        if (sound.downloaded) {
+            if (injector) {
+                injector.stop();
+            }
+            injector = Audio.playSound(sound, {
+                position: position,
+                volume: volume,
+                localOnly: localOnly
+            });
+        }
+    }
+
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+    /* CREATE A MARKER: Checks that marker does not already exist, then calculate position of avatar's hand and 
+    create a marker there */
+    var marker;
+    function createMarker() {
+        if (marker) {
+            return;
+        }
+        var parentJointIndex = MyAvatar.getJointIndex(dominantHandJoint);
+        var localPosition = (dominantHandJoint === "RightHand") ? { x: -0.0011, y: 0.0458, z: 0.0195 } : 
+            { x: -0.0011, y: 0.0458, z: 0.0195 };
+        var localRotation = (dominantHandJoint === "RightHand") ? Quat.fromVec3Degrees({ x: -90, y: 0, z: 0 }) : 
+            Quat.fromVec3Degrees({ x: -90, y: 0, z: 0 });
+        // We'll need a better back up joint for avatars without the required joint in the future
+        // get dimensions based on avatar height
+        marker = Entities.addEntity({
+            name: "Draw App Marker",
+            type: "Model",
+            description: "CC_BY Poly by Google",
+            modelURL: Script.resolvePath("Assets/Models/sharpie.obj"),
+            parentID: MyAvatar.sessionUUID,
+            parentJointName: dominantHandJoint,
+            parentJointIndex: parentJointIndex,
+            localPosition: localPosition,
+            localRotation: localRotation,
+            localDimensions: { x: 0.0447, y: 0.0281, z: 0.1788 }
+        }, 'avatar');
+    }
+
+    /* REGISTER CONTROLLER MAPPING: Listen for controller trigger movements and act when the trigger is pressed or 
+    released */
+    var MINIMUM_TRIGGER_PRESS_VALUE = 0.97;
+    var controllerMapping;
+    var controllerMappingName = 'Hifi-MyControllerMappingName';
```

Comment:
please rename the mapping to something more descriptive - perhaps `hifi-drawApp-controllerMapping` ?